### PR TITLE
Clean up gradle catalogue and use compose BOM

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

Simplification compose version management by using [BOM](https://developer.android.com/jetpack/compose/setup#bom-version-mapping)

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: libraries/core/build.gradle.kts [hunk 1]
```diff
@@ -35,23 +35,27 @@ dependencies {
     api(project(":libraries:customisations"))
     api(libs.kotlin.coroutines.android)
     api(libs.androidx.lifecycle.common)
-    api(libs.compose.animation.core)
-    api(libs.compose.runtime)
+    api("androidx.compose.animation:animation-core")
```
Comment (LachlanMcKee) [inline libraries/core/build.gradle.kts]: Is this the right way to approach this? I'm not a fan of having two different styles here. Does version catalogs not work with BOMs?

You lose the type safety this way

Comment (LachlanMcKee) [reply libraries/core/build.gradle.kts]: I would suggest looking at this solution: https://stackoverflow.com/questions/67192785/gradle-7-0-version-catalog-for-maven-bom

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -8,7 +8,7 @@ accompanist = "0.28.0"
 androidx-lifecycle = "2.5.1"
 androidx-navigation-compose = "2.5.1"
 coil = "2.2.1"
-compose = "1.3.0"
+composeBom = "2023.01.00"
 composeCompiler = "1.3.1"
```
Comment (LachlanMcKee) [inline gradle/libs.versions.toml:12]: Is the compiler version in the BOM somehow?

Comment (KovalevAndrey) [reply gradle/libs.versions.toml:12]: https://developer.android.com/jetpack/compose/setup#why_isnt_the_compose_compiler_library_included_in_the_bom

File: libraries/core/build.gradle.kts (hunk 1 above)
Comment (KovalevAndrey) [reply libraries/core/build.gradle.kts]: I put them back into catalogue

**Review by LachlanMcKee — APPROVED** [review]

**Pull request merged**